cryptofeed/backends/postgres.py

```diff
@@ -37,17 +40,44 @@ def __init__(self, host='127.0.0.1', user=None, pw=None, db=None, table=None, nu
         self.pw = pw
         self.host = host
 
+        self._cache_size = cache_size
+        self._cache = []
+        self._cache_counter = 0
+        self._cache_lock = asyncio.Lock()
+
     async def _connect(self):
         if self.conn is None:
             self.conn = await asyncpg.connect(user=self.user, password=self.pw, database=self.db, host=self.host)
 
     async def write(self, feed: str, pair: str, timestamp: float, receipt_timestamp: float, data: dict):
+        time = dt.utcfromtimestamp(timestamp)
+        rtime = dt.utcfromtimestamp(receipt_timestamp)
+
+        async with self._cache_lock:
```

Comment:
why do you need a lock? There are no awaits in this code block

Reply:
also, you can easily avoid the locking in the case where the cache is not used, so at a minimum the locking should only be in the case where the cache is being utilized. I'm still not convinced its needed for that case though.

Reply:
You are right, there is no need for a lock. I had a previous implementation where it was needed, but didnt think to remove the lock with this current implementation. Changes comitted.